lib/models/device.coffee

```diff
@@ -249,6 +249,41 @@ exports.getApplicationName = (uuid, callback) ->
 	exports.get(uuid).get('application_name').nodeify(callback)
 
 ###*
+# @summary Get application container information
+# @name getApplicationInfo
+# @public
+# @function
+# @memberof resin.models.device
+#
+# @param {String} uuid - device uuid
+# @fulfil {Object} - application info
+# @returns {Promise}
+#
+# @example
+# resin.models.device.getApplicationInfo('7cf02a6').then(function(appInfo) {
+# 	console.log(appInfo);
+# });
+#
+# @example
+# resin.models.device.getApplicationInfo('7cf02a6', function(error, appInfo) {
+# 	if (error) throw error;
+# 	console.log(appInfo);
+# });
+###
+exports.getApplicationInfo = (uuid, callback) ->
+	exports.get(uuid).then (device) ->
```

Comment:
@Page- @lekkas I believe the logic to check the minimum supervisor version and show a warning message should live in the API (the supervisor proxy) instead of here. We're hardcoding a specific minimum supervisor version in the SDK (which its not ideal) and it still doesn't protect users using the API directly, or the Python SDK for example.

If this logic is in the supervisor proxy, the endpoint can return a custom HTTP header to determine a warning notice (e.g: `X-Notice`, or something like that), that all SDKs can use to generically display this.